### PR TITLE
Add option to install Python 3 miniconda

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: bin/deploy [hunk 1]
```diff
@@ -22,21 +23,24 @@ usage() {
         print_error "$0: $1"
     fi
     print_error
-    print_error "Usage: $0 [-b] [-h]"
+    print_error "Usage: $0 [-3] [-b] [-h]"
     print_error
     print_error "Specifc options:"
     print_error "   -b          use bleeding edge conda packages"
+    print_error "   -t          use Python 3"
```
Comment (jhoblitt) [inline bin/deploy]: `-3`

File: bin/deploy [hunk 2]
```diff
@@ -56,23 +60,32 @@ cd "$LSSTSW"
 
 test -f "$LSSTSW/miniconda/.deployed" || ( # Anaconda
     cd sources
+
+    if [[ $USE_PYTHON3 == true ]]; then
+        PYVER_PREFIX=3
+        MINICONDA_VERSION=${MINICONDA3_VERSION}
+    else
+        PYVER_PREFIX=2
+        MINICONDA _VERSION=${MINICONDA2_VERSION}
```
Comment (jhoblitt) [inline bin/deploy]: Typo

Comment (timj) [inline bin/deploy]: THat's confusing because I fixed that last week (otherwise Travis would not work for lsst_build).

**Comment by jhoblitt** [comment]
The conda env files names are used in travis driver script and will need to be updated.  The new `-3` option should also be added to the travis matrix.

**Comment by timj** [comment]
I'm actually wondering if you need to retain the old names of the package list and just add a 3 for the new version. Because otherwise we have to ensure that miniconda2 is updated at the same time as we do this because it needs to know the package listing text file names. There was a nice symmetry to renaming the conda 2 package lists but I don't think we can do it.

**Comment by jhoblitt** [comment]
That is a valid concern, renaming the package file would break the existing tagged versions of miniconda2.

**Comment by timj** [comment]
Obviously the travis -3 checks will fail until lsst/lsst_build#17 is merged.

**Comment by timj** [comment]
Also, @jhoblitt can you build me a quick conda3 linux package list please?

**Comment by jhoblitt** [comment]
```
astropy                   1.1.2               np111py35_0  
cycler                    0.10.0                   py35_0  
cython                    0.24                     py35_0  
fontconfig                2.11.1                        5  
freetype                  2.5.5                         1  
future                    0.15.2                   py35_0  
libgfortran               3.0.0                         1  
libpng                    1.6.17                        0  
libxml2                   2.9.2                         0  
matplotlib                1.5.1               np111py35_0  
nomkl                     1.0                           0  
numpy                     1.11.0             py35_nomkl_1  [nomkl]
openblas                  0.2.14                        4  
openssl                   1.0.2h                        1  
pandas                    0.18.1              np111py35_0  
pip                       8.1.2                    py35_0  
pyparsing                 2.1.4                    py35_0  
pyqt                      4.11.4                   py35_2  
python                    3.5.1                         0  
python-dateutil           2.5.3                    py35_0  
pytz                      2016.4                   py35_0  
qt                        4.8.7                         2  
readline                  6.2                           2  
requests                  2.10.0                   py35_0  
scipy                     0.17.1          np111py35_nomkl_0  [nomkl]
setuptools                22.0.5                   py35_0  
sip                       4.16.9                   py35_0  
six                       1.10.0                   py35_0  
sqlalchemy                1.0.13                   py35_0  
sqlite                    3.13.0                        0  
tk                        8.5.18                        0  
wheel                     0.29.0                   py35_0  
xz                        5.0.5                         1  
zlib                      1.2.8                         3  
```

**Comment by timj** [comment]
@jhoblitt  thanks for the listing but can you do it in `conda list -e` mode please? I also need a pinned list for conda2 on linux where you've done a `conda install future`.

**Comment by timj** [comment]
I added the package list for Linux. I think this is ready to merge. It needs lsst_build to be merged first if we want travis to pass (which will break lsst_build travis because it is relying on this branch).

**Pull request merged**